internal/queryhandler/audit_log_server.go

```diff
@@ -75,7 +81,43 @@ func (s *auditLogServer) GetByDateRange(request *doApi.GetAuditLogByDateRangeReq
 		}
 
 		hre := s.auditFormatter.NewHumanReadableEvent(ctx, e)
+		err = stream.Send(hre)
+		if err != nil {
+			return errors.TranslateToGrpcError(err)
+		}
+	}
 
+	return nil
+}
+
+// GetUserActions returns human readable events caused by the given user actions
+func (s *auditLogServer) GetUserActions(email *wrappers.StringValue, stream doApi.AuditLog_GetUserActionsServer) error {
+	ctx := context.Background()
+
+	user, err := s.userRepo.ByEmail(ctx, email.GetValue())
+	if err != nil {
+		return errors.TranslateToGrpcError(err)
+	}
+	events, err := s.esClient.Retrieve(ctx, &esApi.EventFilter{})
+	if err != nil {
+		return errors.TranslateToGrpcError(err)
+	}
+
+	for {
+		e, err := events.Recv()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return errors.TranslateToGrpcError(err)
+		}
+		// TODO: extend EventFilter and implement on store level? should be cheaper than iterating through all events?
+		// 	- problem: event metadata are stored as json
+		if e.Metadata[auth.HeaderAuthId] != user.Id {
```

Comment:
You're right that the user id is only part of the metadata and you can not filter by that. I think we have to think through options here. I am not sure if going through all events of all time is maybe to costly.

Reply:
yeah, I don't think so as well, and besides that, this system shouldn't encounter too many events throughout its lifecycle and it's not like this will be used every day. As you said last time if this turns out to be costly we can always optimize but if you think we can still optimize here let me know and I can set up a small meeting to brainstorm.

Reply:
after discussing the issue verbally we came up with the following:

- a general restructure for the event structure will happen sooner or later (which should allow for a much better implementation)
- for now, limiting the query to a max 1 year should be enough as a preventive measure until the restructure to filter on the store level is implemented
- All queries are bound to a time-out which will stop the query if it took too long In case there are too many events within the limiting period.